content/docs/08-troubleshooting/01-migration-guide/01-migration-guide-4-0.mdx

```diff
@@ -14,10 +14,18 @@ description: Learn how to upgrade AI SDK 3.4 to 4.0.
 ## Recommended Process
 
 1. [Migrate to AI SDK 3.4](/docs/troubleshooting/migration-guide/migration-guide-3-4).
-2. Resolve all deprecation warnings.
-3. Upgrade to AI SDK 4.0.
-4. _Optional_: Run the [codemod](#codemods) automated migrations.
-5. Follow the breaking changes guide below.
+2. Upgrade to AI SDK 4.0.
```

Comment:
do we want to add in pnpm install ai@latest or 4.0 somwhere?

Reply:
was thinking about how to be do it and opted for the `AI SDK 4.0 package versions` below. we can improve further if you have ideas